test/NonBlockingTests/Program.cs

```diff
@@ -604,7 +465,7 @@ private static void Continuity002()
         {
             var dict = new NonBlocking.ConcurrentDictionary<object, int>();
 
-            Parallel.ForEach(Enumerable.Range(0, 10000),
```

Comment:
These tests are more of stress tests looking for races. I think `Parallel.For` will be faster, but I think it will partition the sequence evenly between threads in large chunks and we might be losing some randomization that Parallel.ForEach provides. 
I think either `Parallel.ForEach` is smart enough to do the same as what `Parallel.For` does, so there is no difference, or it results in a less optimal and more random/stressful pattern. In either case `Parallel.ForEach` would be preferable.

Reply:
Speed is less important for these as they are meant to be run continuously to a failure.

Reply:
Actually, it won't probably even matter in .NET 8 because `RangeIterator` will become `IList<int>`: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Linq/src/System/Linq/Range.SpeedOpt.cs#L10

That change landed just a day after this PR - do you want me to revert this?
The test might need a more hands-on approach with manual threading...my thinking here was that, worst case, `Parallel` might have to enumerate `ForEach` in an SPMC manner causing it to be slower and not stress the impl. enough, but then again, this is yet another assumption which isn't good for multi-threaded code 😅

Reply:
I think `Parallel.ForEach` will try serve increasing chunks of data, but it's been long time since Iooked at that code. 
If Range optimization defeats our randomization aspiration, we might need to implement some kind of "UnoptimizedRange".
 
Let's revert this part. If this can be improved, it may need more thought.